p2p/dial.go

```diff
@@ -197,30 +194,6 @@ func (s *dialstate) newTasks(nRunning int, peers map[discover.NodeID]*Peer, now
 	// Expire the dial history on every invocation.
 	s.hist.expire(now)
 
-	// Create dials for static nodes if they are not connected.
-	for id, t := range s.static {
-		err := s.checkDial(t.dest, peers)
-		switch err {
-		case errNotWhitelisted, errBlacklisted, errSelf:
-			log.Warn("Removing static dial candidate", "id", t.dest.ID, "addr", &net.TCPAddr{IP: t.dest.IP, Port: int(t.dest.TCP)}, "err", err)
-			delete(s.static, t.dest.ID)
-		case nil:
-			s.dialing[id] = t.flags
-			newtasks = append(newtasks, t)
-		}
-	}
-	// If we don't have any peers whatsoever, try to dial a random bootnode. This
-	// scenario is useful for the testnet (and private networks) where the discovery
-	// table might be full of mostly bad peers, making it hard to find good ones.
-	if len(peers) == 0 && len(s.bootnodes) > 0 && needDynDials > 0 && now.Sub(s.start) > fallbackInterval {
-		bootnode := s.bootnodes[0]
-		s.bootnodes = append(s.bootnodes[:0], s.bootnodes[1:]...)
-		s.bootnodes = append(s.bootnodes, bootnode)
-
-		if addDial(dynDialedConn, bootnode) {
-			needDynDials--
-		}
-	}
 	// Use random nodes from the table for half of the necessary
```

Comment:
Only a half of dyndials are made to nodes in the recent lookup buffer. Targets for the other half are randomly chosen from the node table containing 16*256 entries (the entries are checked and refreshed every hour).
Is this okay for our purpose?

Reply:
Oh I didn't realize this is what was happening - I had thought that dyndials were all looking up the RLPX node table. Does this recent lookup buffer overlap significantly with the redial list?